kpops/components/base_components/kafka_connector.py

```diff
@@ -25,50 +20,113 @@
     KafkaConnectorType,
 )
 from kpops.components.base_components.base_defaults_component import deduplicate
+from kpops.components.base_components.helm_app import HelmApp, HelmAppValues
 from kpops.components.base_components.models.from_section import FromTopic
-from kpops.components.base_components.models.resource import Resource
 from kpops.components.base_components.pipeline_component import PipelineComponent
 from kpops.utils.colorify import magentaify
 from kpops.utils.docstring import describe_attr
 
 log = logging.getLogger("KafkaConnector")
 
 
-class KafkaConnector(PipelineComponent, ABC):
-    """Base class for all Kafka connectors.
+class KafkaConnectorResetter(HelmApp):
```

Comment:
I prefer that the kafka_connector.py file starts with the KafkaConnector class, then the sink/source connectors, and, in the end, with the Kafka ConnectorResetter class. WDYT?

Reply:
Is it possible now to define in your pipeline.yaml a `KafkaConnectorResetter`? Because it is a HelmApp, and if I call KPOps to deploy, will it deploy it?

Reply:
I agree, for some reason it slightly changes the pipeline schema when I reorder the classes

```diff
diff --git a/docs/docs/schema/pipeline.json b/docs/docs/schema/pipeline.json
index 186863f6..aebe7127 100644
--- a/docs/docs/schema/pipeline.json
+++ b/docs/docs/schema/pipeline.json
@@ -306,13 +306,13 @@
                 },
                 "type": {
                     "const": "kafka-sink-connector",
+                    "default": "kafka-sink-connector",
                     "title": "Type"
                 }
             },
             "required": [
                 "name",
-                "app",
-                "type"
+                "app"
             ],
             "title": "KafkaSinkConnector",
             "type": "object"
@@ -401,13 +401,13 @@
                 },
                 "type": {
                     "const": "kafka-source-connector",
+                    "default": "kafka-source-connector",
                     "title": "Type"
                 }
             },
             "required": [
                 "name",
-                "app",
-                "type"
+                "app"
             ],
             "title": "KafkaSourceConnector",
             "type": "object"
```

I don't know why it differs from the other components then, would prefer to have it consistent.

Reply:
We _could_ technically make it possible to define `KafkaConnectorResetter` in the pipeline, but what would be the point as it's not meant to be used standalone. Or is there a use-case?
That's why it's excluded from the public components

Reply:
No, there isn't any use-case I can think of so far. I just wanted to know if the scenario is possible or not.